src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3182,6 +3197,20 @@ ImmutableArray<SynthesizedSimpleProgramEntryPointSymbol> buildSimpleProgramEntry
             }
         }
 
+        internal SynthesizedPrimaryConstructor? PrimaryConstructor
+        {
+            get
+            {
+                var declared = Volatile.Read(ref _lazyDeclaredMembersAndInitializers);
+                if (declared is not null && declared != DeclaredMembersAndInitializers.UninitializedSentinel)
+                {
+                    return declared.PrimaryConstructor;
+                }
```

Comment:
Is this necessary, or could we simply return `GetMembersAndInitializers().PrimaryConstructor` always?

Reply:
> Is this necessary, or could we simply return GetMembersAndInitializers().PrimaryConstructor always?

Obtaining this symbol is generally a subject to an infinite loop. I'd rather use all available help to avoid that. Even if today we might not have a scenario making this absolutely necessary.